### PR TITLE
Fix crashing statefulset entities

### LINKED ISSUE #55: Agent crashes if statefulset doesn't have a version label
agent should be able to handle statefulsets without a version label.

```
KeyError: 'version'
ZMON agent failed. Retrying after 60 seconds ...
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/main.py", line 104, in sync
    all_current_entities = discovery.get_entities() + [account_entity]
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/discovery/kubernetes/cluster.py", line 134, in get_entities
    postgresql_cluster_member_entities, postgresql_database_entities))
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/discovery/kubernetes/cluster.py", line 454, in get_cluster_statefulsets
    'version': obj['metadata']['labels']['version']
KeyError: 'version'
2018-02-22 16:04:18,537 ERROR: ZMON agent failed. Retrying after 60 seconds ...
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/main.py", line 104, in sync
    all_current_entities = discovery.get_entities() + [account_entity]
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/discovery/kubernetes/cluster.py", line 134, in get_entities
    postgresql_cluster_member_entities, postgresql_database_entities))
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/discovery/kubernetes/cluster.py", line 454, in get_cluster_statefulsets
    'version': obj['metadata']['labels']['version']
KeyError: 'version'
ZMON agent failed. Retrying after 60 seconds ...
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/main.py", line 104, in sync
    all_current_entities = discovery.get_entities() + [account_entity]
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/discovery/kubernetes/cluster.py", line 134, in get_entities
    postgresql_cluster_member_entities, postgresql_database_entities))
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/discovery/kubernetes/cluster.py", line 454, in get_cluster_statefulsets
    'version': obj['metadata']['labels']['version']
KeyError: 'version'
2018-02-22 16:05:28,838 ERROR: ZMON agent failed. Retrying after 60 seconds ...
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/main.py", line 104, in sync
    all_current_entities = discovery.get_entities() + [account_entity]
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/discovery/kubernetes/cluster.py", line 134, in get_entities
    postgresql_cluster_member_entities, postgresql_database_entities))
  File "/usr/local/lib/python3.6/dist-packages/zmon_agent-0.1-py3.6.egg/zmon_agent/discovery/kubernetes/cluster.py", line 454, in get_cluster_statefulsets
    'version': obj['metadata']['labels']['version']
KeyError: 'version'
```
Introduced in #53

### PR DESCRIPTION
**Pull request by mohabusama**
Fix crashing statefulset entities

Fix #55

### CODE REVIEW

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 1]
```diff
@@ -643,7 +643,7 @@ def get_postgresql_cluster_members(kube_client, cluster_id, alias, environment,
             'region': region,
             'cluster_dns_name': service_dns_name,
             'pod': pod.name,
-            'pod_phase': obj['status']['phase'],
+            'pod_phase': obj.get('status', {}).get('phase', ''),
             'image': container['image'],
             'container_name': container['name'],
             'ip': obj['status'].get('podIP', ''),
```
Comment (avaczi) [inline zmon_agent/discovery/kubernetes/cluster.py]: If `status` is not always there (see 3 lines above), this should also become a `.get()`

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 2]
```diff
@@ -673,37 +673,37 @@ def get_postgresql_databases(cluster_id, alias, environment, region, infrastruct
...
                 'type': POSTGRESQL_DATABASE_TYPE,
                 'kube_cluster': cluster_id,
                 'alias': alias,
                 'environment': environment,
                 'created_by': AGENT_TYPE,
                 'infrastructure_account': infrastructure_account,
                 'region': region,
-                'version': pgcluster['version'],
-                'postgresql_cluster': pgcluster['id'],
+                'version': pgcluster.get('version'),
```
Comment (avaczi) [inline zmon_agent/discovery/kubernetes/cluster.py:684]: I think these are safe - unless there are lurking entities without the keys

File: zmon_agent/discovery/kubernetes/cluster.py (hunk 1 above)
Comment (mohabusama) [reply zmon_agent/discovery/kubernetes/cluster.py]: Done

File: zmon_agent/discovery/kubernetes/cluster.py (hunk 2 above)
Comment (mohabusama) [reply zmon_agent/discovery/kubernetes/cluster.py:684]: Better safe ;)

Comment (avaczi) [reply zmon_agent/discovery/kubernetes/cluster.py:684]: absolutely :)

**Comment by avaczi** [comment]
:+1:

**Comment by mohabusama** [comment]
👍

**Pull request merged**